pkg/proxy/types.go

```diff
@@ -87,6 +90,8 @@ type Endpoint interface {
 	String() string
 	// GetIsLocal returns true if the endpoint is running in same host as kube-proxy, otherwise returns false.
 	GetIsLocal() bool
+	// GetTopology returns the topology information of the endpoint.
+	GetTopology() map[string]string
```

Comment:
`Topology` must be stored in the `Endpoint` so that we can get topology information when we filter the topologically-best set of endpoints.